File: Atividade-8/condicional.c
#include <stdio.h>
#include <stdlib.h>

int CalcularQuadrado(int numero) {
    return numero * numero;
}

int CalcularFatorial(int numero) {
    if (numero == 0) {
        return 1;
    } else {
        return numero * CalcularFatorial(numero - 1);
    }
}

int main() {
    
    int opcao = 0;
    
    while (opcao != 3) {
        printf("1 - Calcule o quadrado de um numero\n");
        printf("2 - Calcule o fatorial de um numero\n");
        printf("3 - Sair\n");

        scanf("%d", &opcao);

        switch (opcao) {
            case 1: {
                int numero;
                printf("Digite um numero: ");
                scanf("%d", &numero);
                int quadrado = CalcularQuadrado(numero);
                printf("O quadrado de %d e %d\n", numero, quadrado);
                break;
            }
            case 2: {
                int numero;
                printf("Digite um numero: ");
                scanf("%d", &numero);
                int fatorial = CalcularFatorial(numero);
                printf("O fatorial de %d e %d\n", numero, fatorial);
                break;
            }
            case 3:
                printf("Saindo...\n");
                break;
            default:
                printf("Opcao invalida\n");
        }

    }

    return 0;
}

File: Atividade-8/condicional.py
def calcular_quadrado(numero):
    return numero * numero

def calcular_fatorial(numero):
    if numero == 0 or numero == 1:
        return 1
    else:
        return numero * calcular_fatorial(numero - 1)

def main():
    
    escolha = -1
    while escolha != 3:
        print("1 - Calcular quadrado de um número")
        print("2 - Calcular fatorial de um número")
        print("3 - Sair")

        escolha = int(input("Escolha uma opção: "))

        if escolha == 1:
            num = int(input("Digite um número para calcular o quadrado: "))
            resultado = calcular_quadrado(num)
            print(f"O quadrado de {num} é {resultado}")
        elif escolha == 2:
            num = int(input("Digite um número para calcular o fatorial: "))
            resultado = calcular_fatorial(num)
            print(f"O fatorial de {num} é {resultado}")
        elif escolha == 3:
            print("Saindo...")
        else:
            print("Opção inválida. Tente novamente.")

main()